File: RemoveDuplicatesFromList/RemoveDuplicatesFromList.cpp
#include <bits/stdc++.h>
#include "Solution.cpp"

using namespace std;


int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
        if(i > 0) {
            assert(list[i] >= list[i-1]);
        }
    }
    ListNode* head = ListNode().createList(list);

    ListNode* result = removeDuplicatesFromList(head);
    vector<int> resultList = ListNode().extractList(result);
    
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}

File: RemoveDuplicatesFromList/RemoveDuplicatesFromList.java
import java.io.*;
import java.util.*;
import crio.ds.List.ListNode;

class RemoveDuplicatesFromList {
	public static void main(String [] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		ArrayList<Integer> list = new ArrayList<Integer>();
		for ( int i = 0; i < n; i++) {
			int x = sc.nextInt();
			list.add(x);
		}

		ListNode head = ListNode.createList(list);
		head = Solution.removeDuplicatesFromList(head);
		List<Integer> result = ListNode.extractList(head);
		for ( int i = 0; i < result.size(); i++) {

			System.out.print(result.get(i) + " ");
		}
	}
}

